src/drive/lib/RouterContext.jsx

```diff
@@ -0,0 +1,14 @@
+import React from 'react'
+import { withRouter } from 'react-router'
+
+export const RouterContext = React.createContext()
+
+export const RouterContextProvider = withRouter(
```

Comment:
> I think we could go a step further by directly exposing useRouter

👍 

> I think if intermediate children are not accessing the context, they will not be rendered, so that should be OK.

This was my thinking as well, only consumers will be re-rendered when the router changes.